pxtlib/melody-editor/melodyArray.ts

```diff
@@ -0,0 +1,125 @@
+namespace pxtmelody {
+    export class MelodyArray {
+        // check that array is 8x8 
+        private tempo: number = 120;
+        private width: number = 8;
+        private height: number = 8;
+        private melody: boolean[][];
+        private title: string = "Name this tune";
+
+        // constructor
+        constructor(tempo?: number) {
+            if (tempo) this.tempo = tempo;
+            // set all elements to false
+            this.melody = new Array(this.width);
+            for (let i = 0; i < this.width; i++) {
+                this.melody[i] = new Array(this.height).fill(false);
+            }
+        }
+
+        public setTitle(title: string): void {
+            this.title = title;
+        }
+
+        public setTempo(tempo: number): void {
+            this.tempo = tempo;
+        }
+
+        public getArray(): boolean[][] {
+            return this.melody;
+        }
+
+        public setArray(array: boolean[][]): void {
+            this.melody = array;
+        }
+
+        public getColor(row: number): number {
+            // TODO
+            return 0;
+        }
+
+        public getValue(row: number, col: number): boolean {
+            return this.melody[row][col];
+        }
+
+        public getWidth(): number {
+            return this.width;
+        }
+
+        public getHeight(): number {
+            return this.height;
+        }
+
+        public updateMelody(row: number, col: number) {
+            this.melody[row][col] = !this.melody[row][col];
+        }
+
+
+        // function to turn into string
+        public getStringRepresentation(): string {
+            let stringMelody: string = "";
+            let queues: string[][];
```

Comment:
nitpick: for `stringMelody` and `queues`, assign the variable in the delcaration (e.g. `let queues: string[][] = new Array(this.width)`)